packages/passport-interface/src/SubscriptionManager.ts

```diff
@@ -279,19 +322,32 @@ export class FeedSubscriptionManager {
       );
     }
 
-    // Did the caller pass the wrong type of credential?
-    if (info.credentialType && info.credentialType !== credential?.type) {
+    // If the required credential type is "email-pcd" then we *must* receive
+    // an email-pcd as a parameter
+    if (info.credentialType === "email-pcd" && credential === undefined) {
       throw new Error(
-        credential?.type
-          ? `wrong credential type "${credential.type}" (expected "${info.credentialType}") on ${providerUrl} feed ${info.id}`
-          : `missing credential of type "${info.credentialType} on ${providerUrl} feed ${info.id}`
+        `email-pcd credential required but did not receive an email-pcd`
       );
     }
+
+    // If the required credential type is semaphore-signature-pcd, then we
+    // must *not* receive a credential parameter here, since we will always
+    // generate a new semaphore signature
+    if (
+      info.credentialType === "semaphore-signature-pcd" &&
+      credential &&
+      credential.type !== "semaphore-signature-pcd"
```

Comment:
I would use the constants from the packages rather than hardcoding them here.

Reply:
This is one of those cases where TypeScript already knows what the valid string values are, and VSCode even autocompletes them for you. I'll change it to use the constants though.